Week3/todo-app/api/getReminders.js

```diff
@@ -0,0 +1,62 @@
+const logger = require("../logger");
+const dbManager = require("../db");
+const user = require("../user");
+const { successResponse, failureResponse } = require("../response");
+const { responseObject } = require("../config");
+
+const getReminders = (req, res) => {
+  responseObject.operation = "getReminders";
+  const userCredentials = {};
+  userCredentials.id = parseInt(req.headers.userid, 10);
+  userCredentials.username = req.headers.username;
+  logger.log(`GET: /list/remind, headers: ${JSON.stringify(userCredentials)}`);
+  if (!userCredentials) {
+    responseObject.message =
+      "user credentials not provided with the headers: userId, username.";
+    logger.log(responseObject.message, false);
+    res.statusCode = 403;
+    res.send(failureResponse(responseObject));
+  } else if (
+    userCredentials.id !== user.id ||
+    userCredentials.username !== user.username
+  ) {
+    responseObject.message =
+      "user credentials do not match with the logged in user.";
+    logger.log(responseObject.message, false);
+    res.statusCode = 401;
+    res.send(failureResponse(responseObject));
+  } else {
+    dbManager
+      .query(
+        "SELECT" +
+          " TodoList.userID as userId," +
+          " TodoList.id as TodoListId," +
+          " TodoList.name as TodoListName," +
+          " TodoList.description as TodoListDescription," +
+          " Reminder.id as ReminderId," +
+          " Reminder.description as ReminderDescription," +
+          " Reminder.remindingTime as RemindingTime" +
+          " FROM TodoList INNER JOIN Reminder" +
+          " ON TodoList.id = Reminder.todoListID" +
+          " WHERE userID = ?",
+        user.id
+      )
+      .then(queryResult => {
+        responseObject.message =
+          "Successfully fetched the todo lists with reminders.";
+        logger.log(`${responseObject.message}: ${JSON.stringify(user)}`);
+        res.send({
+          ...successResponse(responseObject),
+          data: queryResult
+        });
+      })
+      .catch(err => {
+        responseObject.message = `Query Error occurred. ${err.message}`;
+        logger.log(responseObject.message, false);
+        res.statusCode = 500;
```

Comment:
In this case, HTTP codes are most likely not going to change. However, it's usually a good idea to have them as constants on a separate file that we can import. This way, if we made a typo and we wrote 305 instead of 500, changing this value should be one line change.

Reply:
You are absolutely right about that, I just noticed that it may cause problems. Thanks.